src/main/java/uk/nhs/hee/tis/trainee/forms/service/FormRPartAService.java

```diff
@@ -256,24 +256,14 @@ public Optional<FormRPartADto> unsubmitFormRPartAById(UUID id) {
     log.info("Request to unsubmit FormRPartA with id : {}", id);
 
     return repository.findById(id)
-        .map(this::unsubmit)
+        .map(form -> {
+          form.setLifecycleState(LifecycleState.UNSUBMITTED);
+          repository.save(form);
+          log.info("Unsubmitted successfully for trainee {} with form Id {} (FormRPartA)",
+              form.getTraineeTisId(), form.getId());
+          return form;
```

Comment:
You're returning the original form here instead of the updated one.
It's a minor issue right now, but this could cause issues in the future when we start to record status history.